explorer/syntax/lexer.lpp

```diff
@@ -86,6 +86,7 @@ LESS_LESS            "<<"
 LET                  "let"
 LIBRARY              "library"
 MATCH                "match"
+MATCH_FIRST          "match_first"
```

Comment:
How exploratory is this? Would it make sense to tag this as `__match_first`, similar to what we've done for other experimental features (like `__mix`)? Or is this more final?

Reply:
The keyword [appears in the design](https://github.com/carbon-language/carbon-lang/blob/trunk/docs/design/generics/details.md#prioritization-rule) and was added there by an approved proposal, but it's in an "open questions" section. I don't have a strong view here. If you'd prefer that keywords that haven't been adopted by a proposal are all named with a `__` then I'll do that.

Reply:
Per discussion -- let's use `__match_first`, since `match_first` is in examples, but `match_first` is somewhat confusing given there's also `match` (which also matches the first) and may not be the final keyword name as a consequence.